### PR TITLE
Add typings for TypeScript interop

### PR DESCRIPTION
**Pull request by rand0me**
Hi, Luke
Epic thanks for this wonderful alternative to default GA client!

I use this package with TypeScript and while it hasn't typings I forced to copy-paste my types.d.ts from project to project. It would be great if you merge this PR to make this awesome library support TypeScript.

Thanks!

### CODE REVIEW

File: ganalytics.d.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+interface IGAnalyticsOptions {
```
Comment (lukeed) [inline ganalytics.d.ts]: Let's export this

File: ganalytics.d.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    av?: string;
+
+    /**
+     * Indicates the data source type of the hit; eg web or app.
+     * See [Data Source](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#ds).
+     */
+    ds?: string;
+}
+
+interface IGAnalytics {
```
Comment (lukeed) [inline ganalytics.d.ts]: Let's also export this

File: ganalytics.d.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    send: (kind: "pageview" | "event", params?: any) => void;
+}
+
+declare namespace ganalytics {
+    /**
+     * @param trackerID Your Google Analytics tracker ID; eg UA-XXXXXXXX-X
+     * @param options Options
+     * @param toWait When truthy, a pageview event will not be sent immediately upon initialization.
+     */
+    function GAnalyticsFactory(trackerID: string, options?: IGAnalyticsOptions, toWait?: boolean): IGAnalytics;
```
Comment (lukeed) [inline ganalytics.d.ts]: I _think_ this can be `options?: Partial<IGAnalyticsOptions>` ?

File: ganalytics.d.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+    /**
+     * Indicates the data source type of the hit; eg web or app.
+     * See [Data Source](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#ds).
+     */
+    ds?: string;
+}
+
+interface IGAnalytics {
+    send: (kind: "pageview" | "event", params?: any) => void;
```
Comment (lukeed) [inline ganalytics.d.ts]: There's actually a lot more _kinds_ that can be sent. Here's the list: https://github.com/lukeed/ganalytics#type

So let's add a new Enum here: 

```ts
send: (kind: EventType, params?: object) => void
```

> Also, `params` can only be an object.

File: ganalytics.d.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+     * Specifies the application verison.
+     * See [Version](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#av).
+     */
+    av?: string;
+
+    /**
+     * Indicates the data source type of the hit; eg web or app.
+     * See [Data Source](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#ds).
+     */
+    ds?: string;
```
Comment (lukeed) [inline ganalytics.d.ts]: With the `Partial` keyword, we don't have to add `?` to every key

**Comment by lukeed** [comment]
The additional exports are because I know that someone will undoubtedly ask for them, so we might as well jump the gun here 😆 

As for general setup, I don't think we necessarily need the namespace? Might be mistaken here. But GAnalytics will work with/without `new` keyword, so technically, I don't think a factory need be returned at all.

```ts
export type EventType = 'send' | 'pageview' | ...;

export interface Options {
  // ...
}

export interface GAnalytics {
  send: (type: EventType, params?: object) => void;
}

declare const ganalytics: (trackerID: string, options?: Partial<Options>, toWait?: boolean): GAnalytics;

export default ganalytics;
```

**Comment by rand0me** [comment]
Hi, @lukeed ! :wave: 
I just updated typings according to your review and also did overloads for `send()` method, so it could be easily used with different event types utilizing params interfaces.

**Comment by rand0me** [comment]
Hi @lukeed! :wave:

How about merging this PR?
I will be very thankful if you accept it! :)

**Comment by lukeed** [comment]
Hi! So sorry 🙈  thank you for the ping!

This is super awesome & very very much appreciated 😄 I was using it today and found some errors and incomplete sections, but I will merge this as is ATM. I'll apply the fixes soon after – likely today. I have a bunch of them already locally.

Thank you thank you 🙌

**Comment by rand0me** [comment]
Thanks a bunch!
Very kind from you!

**Pull request merged**